src/FormattedText/FormattedText.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import { props, t } from '../utils';
+import flattenDeep = require('lodash/flattenDeep');
+import tlds from 'tlds';
+import { LinkifyIt } from 'linkify-it';
+
+const linkify: LinkifyIt = require('linkify-it')().tlds(tlds);
+
+export type FormattedTextProps = {
+  /** The content of the paragraph */
+  children: string,
+  id?: string,
+  className?: string,
+  style?: React.CSSProperties
+}
+
+export const Props = {
+  children: t.String,
+  id: t.maybe(t.String),
+  className: t.maybe(t.String),
+  style: t.maybe(t.Object)
+};
+
+@props(Props)
+export default class FormattedText extends React.PureComponent<FormattedTextProps> {
+
+  linkify(string: string): (string | JSX.Element)[] {
+    const matches = linkify.match(string);
+    if (matches) {
+      return matches.reduce((acc, match, i) => {
+        const nextMatch = matches[i + 1];
+
+        return [
+          ...(acc || [string.slice(0, match.index)]),
+          <a href={match.url} target='_blank'>{match.raw}</a>,
+          string.slice(match.lastIndex, nextMatch ? nextMatch.index : undefined)
+        ];
+      }, null) || [];
+    } else {
+      return [string];
+    }
+  }
+
+  replaceLinksWithA(children: (string | JSX.Element)[]): (string | JSX.Element)[] {
+    return flattenDeep<string | JSX.Element>(children.map((child) => {
+      if (t.String.is(child)) {
+        return this.linkify(child as string);
```

Comment:
dovrebbe essere safe fare `as string`, visto che abbiamo appena checkato

Reply:
ma perché serve? TS dovrebbe saperlo a questo punto perché `t.String.is` è un type predicate.

Reply:
vedi https://github.com/gcanti/tcomb/blob/master/index.d.ts#L6

Reply:
hai ragione tu :P

Reply:
confermo che compila anche senza il cast